source/features/prevent-pr-commit-link-loss.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import AlertIcon from 'octicon/alert.svg';
+import * as pageDetect from 'github-url-detection';
+import * as textFieldEdit from 'text-field-edit';
+
+import features from '../libs/features';
+import {prCommitRegex, preventPrCommitLinkBreak} from '../libs/utils';
+
+function handleButtonClick(event: delegate.Event<MouseEvent, HTMLButtonElement>): void {
+	const field = event.delegateTarget.form!.querySelector('textarea')!;
+	textFieldEdit.set(field, preventPrCommitLinkBreak(field.value));
+	event.delegateTarget.parentElement!.remove();
+}
+
+function handleInput(event: delegate.Event<InputEvent, HTMLTextAreaElement>): void {
+	const field = event.delegateTarget;
+	const formWarningExists = select.exists('.rgh-fix-pr-commit-links', field.form!);
+
+	if (prCommitRegex.test(field.value) && !formWarningExists) {
+		return select('.form-actions', field.form!)!.prepend(
+			<div className="flash flash-warn mb-2 rgh-fix-pr-commit-links-container">
+				<AlertIcon/>Your PR Commit link may be <a target="_blank" rel="noopener noreferrer" href="https://github.com/sindresorhus/refined-github/issues/2327">misinterpreted by GitHub.</a>
+				<button type="button" className="btn btn-sm primary flash-action rgh-fix-pr-commit-links">Fix link</button>
+			</div>
+		);
+	}
+
+	if (!prCommitRegex.test(field.value) && formWarningExists) {
+		return select('.rgh-fix-pr-commit-links-container', field.form!)!.remove();
+	}
+}
```

Comment:
Can you copy this format? https://github.com/sindresorhus/refined-github/blob/4b08f2421fafc39b19214c0b9a9ecb4d04d824c3/source/features/sync-pr-commit-title.tsx#L42-L56

I think:

```suggestion
function getUI(form: HTMLFormElement): HTMLElement {
	return select('.rgh-fix-pr-commit-links-container') ?? (
		<div className="flash flash-warn mb-2 rgh-fix-pr-commit-links-container">
			<AlertIcon/>Your PR Commit link may be <a target="_blank" rel="noopener noreferrer" href="https://github.com/sindresorhus/refined-github/issues/2327">misinterpreted by GitHub.</a>
			<button type="button" className="btn btn-sm primary flash-action rgh-fix-pr-commit-links">Fix link</button>
		</div>
	);
}

function handleInput(event: delegate.Event<InputEvent, HTMLTextAreaElement>): void {
	const field = event.delegateTarget;
	const notice = getUI(field.form!);
	
	if (prCommitRegex.test(field.value)) {
		select('.form-actions', field.form!).after(notice);
	} else {
		notice.remove();
	}
}
```